core/src/main/resources/reference.conf

```diff
@@ -430,6 +430,17 @@ cassandra-query-journal {
     # Verbose debug logging for offset updates for events by tag queries. Any logging that is per event is
     # affected by this flag. Debug logging that is per query to Cassandra is enabled if DEBUG logging is enabled
     verbose-debug-logging = false
+
+    # Set to a duration to have events by tag queries drop state about persistence ids
+    # this can be set to reduce memory usage of events by tag queries for use cases where there
+    # are many short lived persistence ids
+    # If the metadata for a persistence-id is dropped and the persistence-id is encoutered it will
+    # add a delay of the new-persistence-id-scan-timeout before delivering any new events
+    # If the query is started with NoOffset then the the stream will fail
```

Comment:
why is that?

Reply:
For offset queries we make a best guess which is the first sequence nr for each pid by searching at the start and for doing a short search when we find a new persistence id.

For queries that start from NoOffset we validate that we see every sequence number from 1. If we drop the metadata and then get an event then we can't remain this strict.

The alternative would be to offer the same guarantees for NoOffset queries as we do for Offset queries once we've dropped metadata

Reply:
Actually that's not true. Because it drops into the new persistence id search as the metadata has been dropped it'll do a short search then continue. Removing comment.